pkg/controller/servicebindingrequest/servicecontext_test.go

```diff
@@ -169,5 +169,32 @@ func TestFindOwnedResourcesCtxs_Secret(t *testing.T) {
 		)
 		require.NoError(t, err)
 		require.NotEmpty(t, ownedResourcesCtxs)
+		require.EqualValues(t, 1, len(ownedResourcesCtxs))
+		require.Equal(t, "test_database", ownedResourcesCtxs[0].Service.Object["metadata"].(map[string]interface{})["name"].(string))
 	})
+
+	t.Run("multiple secrets", func(t *testing.T) {
```

Comment:
since we are adding a new test here, we could definitely make some nice table tests, to avoid code copy/pasting.

Reply:
@pedjak I was wondering how would we have table driven tests for this scenario since both the tests has different test logic. implementation of table driven tests should be there when tests uses same logic and but different parameters.

Reply:
well,  the setup for both tests are pretty much similar, the same steps are created for both secrets:
```
secret := mocks.SecretMock("test", "test_database")
	us := &unstructured.Unstructured{}
	uc, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&secret)
	require.NoError(t, err)
	us.Object = uc
	us.SetOwnerReferences([]metav1.OwnerReference{reference})
	route, err := runtime.DefaultUnstructuredConverter.ToUnstructured(mocks.RouteCRMock("test", "test"))
	require.NoError(t, err)
	usRoute := &unstructured.Unstructured{Object: route}
	usRoute.SetOwnerReferences([]metav1.OwnerReference{reference})
	f.S.AddKnownTypes(pgv1alpha1.SchemeGroupVersion, &pgv1alpha1.Database{})
	f.S.AddKnownTypes(routev1.SchemeGroupVersion, &routev1.Route{})
	f.AddMockResource(cr)
	f.AddMockResource(us)
	f.AddMockResource(&unstructured.Unstructured{Object: route})
```

in the new test you have in essence copied the above block and renamed a few variables, but is really different is that the first tests has as input one secret named `test_database`, and the new test has two secrets. The above block could be iterated of a list of input params. Hence, the table test struct could look like:

```
tests := []struct {
		name                string
		secretNames    []string
	}{
  name: "existing selector",
  secretNames: { "test_database" },
},
{
  name: "multiple secrets",
  secretNames: { "test_database", "test_database2" },
}
```

Reply:
Thanks @pedjak makes sense 
pushed changes accordingly here https://github.com/redhat-developer/service-binding-operator/pull/489/commits/5e0e1aa1a283fdd224024b08bca0750166d397fc